Formula/h/hiramu-cli.rb

```diff
@@ -0,0 +1,18 @@
+class HiramuCli < Formula
+ 
+    desc "A cli that interacts with large language models, specifically AWS Bedrock Anthropic's Claude models (Haiku, Sonnet and Opus), to generate text based on prompts."
+    version "v0.1.14"
+    homepage "https://github.com/raphaelmansuy/hiramu-cli"
+    url "https://github.com/raphaelmansuy/hiramu-cli/archive/v#{version}.tar.gz"
+    sha256 "06dd130228cb5f7798b2c3344bf5b1e1b1df5daa40f5893bb4a79c015f8ab55c"
+    license "Apache-2.0"
+  
+    depends_on "rust" => :build
+  
+    def install
+        system "cargo", "install", *std_cargo_args
+        generate_completions_from_executable(bin/"hiramu-cli", "completions")
+      end
+  
+
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.